core/quivr_core/processor/implementations/megaparse_processor.py

```diff
@@ -74,26 +74,42 @@ def processor_metadata(self):
         }
 
     async def process_file_inner(self, file: QuivrFile) -> list[Document]:
-        api_key = str(os.getenv("MEGAPARSE_API_KEY"))
-        megaparse = MegaParseSDK(api_key)
         logger.info(f"Uploading file {file.path} to MegaParse")
-        data = {
-            "method": self.megaparse_config.method,
-            "strategy": self.megaparse_config.strategy,
-            "check_table": self.megaparse_config.check_table,
-            "parsing_instruction": self.megaparse_config.parsing_instruction,
-            "model_name": self.megaparse_config.model_name,
-        }
-        response = await megaparse.file.upload(
-            file_path=str(file.path),
-            **data,
-        )
+        parser = UnstructuredParser(**self.megaparse_config.model_dump())
+        megaparse = MegaParse(parser)
+        response = await megaparse.aload(file.path)
+        logger.info(f"File :  {response}")
         document = Document(
-            page_content=response["result"],
+            page_content=response,
```

Comment:
We still have a bug of not having chunk_size in the metadata if content is smaller. We should always call the splitter the overhead is minimal for small chunks

Reply:
Done